### ISSUE TITLE
Trade communities seem to get bigger then for some cases of beta

### ISSUE THREAD

**Issue by florianjehn** [allfed/pytradeshifts#38]
Comment by Constantin Arnscheidt: "I played around with the code a bit, and it's pretty nice. One comment I have is about the distance trade penalty. I wanted to see what happens in extreme cases. In the most extreme case (e.g. beta = -100), trade communities become really small and localised, which makes sense. However, in a more intermediate scenario (beta=-20), the communities actually look geographically wider than in the default case. (Figures attached.) This seems interesting and unexpected, and is probably worth understanding."

![beta_20](https://github.com/allfed/pytradeshifts/assets/7868527/6e3646b8-208d-4129-a468-0185f0b9ae9c)

**Comment by florianjehn** [comment]
Maybe this is happening as the trade flows overall are low, but similar, so everything gets grouped in one big community?

**Comment by lukaszgajewski** [comment]
### TLDR;
The real-world values of the trade relations exponent is roughly in range [1.2, 1.6]. Setting it as high as 20 or a 100 is perhaps entertaining, but not particularly informative. The code is working as expected.

We should perhaps expand the description of the parameter to include:
- information from apply_distance_cost() doc
- suggestion of reasonable values range

### PyTradeShifts.beta >= 0, trading is harder

![pos_beta](https://github.com/allfed/pytradeshifts/assets/13658920/cd61e02a-a64b-4bef-ad83-99c218f66bbd)
Trade matrix total sum values:

[183034498.2645593, 81893.90255204977, 919.1728082111451, 1.0607289178664083, 3.694565224600827e-05, 4.598923723596655e-12, 3.439801524354341e-23, 1.833181488468943e-41, 7.862603774187596e-72, 1.8097050746552404e-122]

So, the higher the beta the more communities we find, which is as expected. At a point where it flattens out, the results are only due to some residual data left in the matrix.
Essentially when beta is very large it no longer makes sense to talk about any trade, and eventually (e.g., beta=100 for me) the Louvain algorithm breaks because the matrix is all zeroes.

### PyTradeShifts.beta <= 0, trading is easier
![neg_beta](https://github.com/allfed/pytradeshifts/assets/13658920/55dc0a4d-5abd-40fc-bd09-3ab9fcb3692b)
![neg_beta_small](https://github.com/allfed/pytradeshifts/assets/13658920/0465548c-cacf-4dd7-a858-24600d94cb24)

Trade matrix total sum values (first plot):

[183034498.2645593, 1025048857042.2034, 414800451723520.6, 1.1383126159161655e+19, 3.995711941497216e+26, 2.566207248692525e+39, 1.2529575333664141e+61, 4.377335244276899e+97, inf]

(second plot):
[183034498.2645593, 1025048857042.2034, 13859637100616.441, 414800451723520.6, 3.498718063206667e+16, 1.1383126159161655e+19, 2.14806897872408e+22, 3.995711941497216e+26, 1.460112574509737e+32, 2.566207248692525e+39, 6.900006530077103e+48]

So, in the very beginning the number of found communities decreases as expected, then it goes back up because the trade matrix values shoot up, and eventually it explodes as the values explode to infinity, rendering the community detection, again, a meaningless result.

### Code for reproducibility
```python
from src.model import PyTradeShifts
import matplotlib.pyplot as plt
import numpy as np

beta = [0] + list(np.logspace(0, 2, 10))
c_len = []
m_sum = []
plot_betas = []
for b in beta:
    try:
        m = PyTradeShifts(
            crop="Wheat",
            base_year=2018,
            beta=b,
            cd_kwargs={"seed": 2},
            make_plot=False,
            keep_singletons=True,
        )
        c_len.append(len(m.trade_communities))
        m_sum.append(m.trade_matrix.sum().sum())
        plot_betas.append(b)
    except Exception as ex:
        print("it broke")
        print(ex)

_, axs = plt.subplots(2, 1, tight_layout=True, sharex=True, figsize=(5, 5))
axs[0].plot(plot_betas, c_len, "o-")
axs[1].plot(plot_betas, m_sum, "o-")
axs[1].set_xlabel("beta")
axs[0].set_ylabel("# communities")
axs[1].set_ylabel("trade matrix total sum")
plt.savefig("pos_beta.png", bbox_inches="tight")

print(c_len)
print(m_sum)
```

**Comment by florianjehn** [comment]
This makes sense. Thanks for the explanation :)
